core/io/shared/new.rb

```diff
@@ -208,6 +208,26 @@
     @io.internal_encoding.to_s.should == 'IBM866'
   end
 
+  it "does not use binmode argument when mode encoding is specified" do
```

Comment:
**suggestion:**

```suggestion
  it "does not use binary encoding when mode encoding is specified along with binmode: true option" do
```

Actually it uses the `binmode` argument and `#binmode?` will return a specified value. But it doesn't affect io encoding.

There is a test for `"b"` mode and `internal_encoding`. Shouldn't we have a similar case with `binmode`/`textmode` options as well?